File: goraceloty-front-end/src/app/entity/OfferReservation.ts
export class OfferReservation {
  constructor(reservationId: number, startDate: string, numAdult: number, numChildren: number, numOfDays: number, reservationTime: string, adjustedPrice: number, offerID: number, hotelID: number, transportID: number, isPaid: boolean) {
    this.reservationID = reservationId;
    this.startDate = startDate;
    this.numAdult = numAdult;
    this.numChildren = numChildren;
    this.numOfDays = numOfDays;
    this.reservationTime = reservationTime;
    this.adjustedPrice = adjustedPrice;
    this.offerID = offerID;
    this.hotelID = hotelID;
    this.transportID = transportID;
    this.isPaid = isPaid
  }
  reservationID: number;
  startDate: string;
  numAdult: number;
  numChildren: number;
  numOfDays: number;
  reservationTime: string;
  adjustedPrice: number;
  offerID: number;
  hotelID: number;
  transportID: number;
  isPaid: boolean;

  static countHotelOccurrences(reservations: OfferReservation[]): { [key: number]: number } {
    return reservations.reduce((acc, reservation) => {
      acc[reservation.hotelID] = (acc[reservation.hotelID] || 0) + 1;
      return acc;
    }, {} as { [key: number]: number });
  }

  static countTransportOccurrences(reservations: OfferReservation[]): { [key: number]: number } {
    return reservations.reduce((acc, reservation) => {
      acc[reservation.transportID] = (acc[reservation.transportID] || 0) + 1;
      return acc;
    }, {} as { [key: number]: number });
  }
}

File: api-gateway/src/main/java/com/goraceloty/apigateway/travel_agency/entity/OfferReservation.java
package com.goraceloty.apigateway.travel_agency.entity;

import java.sql.Timestamp;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class OfferReservation {
    private Long reservationID;
    private LocalDate startDate;
    private Integer numAdult;
    private Integer numChildren;
    private Integer numOfDays;
    private Timestamp reservationTime;
    private Double adjustedPrice;
    private Long offerID;
    private Long hotelID;
    private Long transportID;
    private Boolean isPaid;
    private Long clientID;
}

